### PR TITLE
Fix trailing slash server url not working

### PR DESCRIPTION
**Pull request by ddshd**
Having a server URL with a trailing slash results in calling endpoints with double `/`.

`localhost/` -> `localhost//api`

Remove trailing slash from server URL before creating URL with endpoint

### CODE REVIEW

**Comment by paulgessinger** [comment]
Hey @ddshd, 

I was under the impression that trailing slashes were already removed when the URL is being saved during the login process, effectively sanitizing it then and there. 

There's no reason not to redo this anyway when making endpoint URLs I suppose. 

Did you do anything special to trigger this or did the login screen just let you add a URL with a trailing slash?

**Comment by ddshd** [comment]
> Hey @ddshd,
> 
> I was under the impression that trailing slashes were already removed when the URL is being saved during the login process, effectively sanitizing it then and there.
> 
> There's no reason not to redo this anyway when making endpoint URLs I suppose.
> 
> Did you do anything special to trigger this or did the login screen just let you add a URL with a trailing slash?

Nothing special, I attached a screen recording with my changes reverted

https://github.com/paulgessinger/swift-paperless/assets/17243132/d56d2bcd-445a-4537-9a77-34aac469fe38

```
Fetching request data for http://192.168.1.200:8050//api/ui_settings/
Getting backend versions
Making API endpoint URL with http://192.168.1.200:8050/ for /api
URL for Endpoint /api: http://192.168.1.200:8050//api/
[...]
Fetching request data for http://192.168.1.200:8050//api/
Checking response of url http://192.168.1.200:8050//api/ui_settings/
```

File: swift-paperless/Networking/Endpoint.swift [hunk 1]
```diff
@@ -208,7 +208,19 @@ extension Endpoint {
     }
 
     func url(url: URL) -> URL? {
-        var result = url.appending(path: path, directoryHint: .isDirectory)
+        // Break down the URL into components
+        guard var components = URLComponents(url: url, resolvingAgainstBaseURL: false) else {
+            return nil
+        }
+        
+        // Remove leading slashes from the path
```
Comment (paulgessinger) [inline swift-paperless/Networking/Endpoint.swift:216]:
```suggestion
        // Remove trailing slashes from the path
```

File: swift-paperless/Networking/Endpoint.swift [hunk 2]
```diff
@@ -208,7 +208,19 @@ extension Endpoint {
     }
 
     func url(url: URL) -> URL? {
-        var result = url.appending(path: path, directoryHint: .isDirectory)
+        // Break down the URL into components
+        guard var components = URLComponents(url: url, resolvingAgainstBaseURL: false) else {
+            return nil
+        }
+
```
Comment (paulgessinger) [inline swift-paperless/Networking/Endpoint.swift:215]:
```suggestion

```

File: swift-paperless/Networking/Endpoint.swift [hunk 3]
```diff
@@ -208,7 +208,19 @@ extension Endpoint {
...
-        var result = url.appending(path: path, directoryHint: .isDirectory)
+        // Break down the URL into components
+        guard var components = URLComponents(url: url, resolvingAgainstBaseURL: false) else {
+            return nil
+        }
+        
+        // Remove leading slashes from the path
+        components.path = components.path.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
+        
+        if (components.url == nil) {
```
Comment (paulgessinger) [inline swift-paperless/Networking/Endpoint.swift:219]:
```suggestion
        if components.url == nil {
```

File: swift-paperless/Networking/Endpoint.swift [hunk 4]
```diff
@@ -208,7 +208,19 @@ extension Endpoint {
...
+            return nil
+        }
+
```
Comment (paulgessinger) [inline swift-paperless/Networking/Endpoint.swift:222]:
```suggestion

```

File: swift-paperless/Networking/Endpoint.swift [hunk 5]
```diff
@@ -208,7 +208,17 @@ extension Endpoint {
...
 
     func url(url: URL) -> URL? {
-        var result = url.appending(path: path, directoryHint: .isDirectory)
+        // Break down the URL into components
+        guard var components = URLComponents(url: url, resolvingAgainstBaseURL: false) else {
+            return nil
+        }
+        // Remove trailing slashes from the path
+        components.path = components.path.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
+
```
Comment (paulgessinger) [inline swift-paperless/Networking/Endpoint.swift:217]:
```suggestion

```

**Pull request merged**